### ISSUE TITLE
Two consumers with same group id connects to same partition after restart

### ISSUE THREAD

**Issue by dannyd30** [tulios/kafkajs#1122]
**Describe the bug**
I have a topic with two partitions Partition -0 and Partition -1.

I ran two consumers with same group id. Consumer 0 subscribes to partition 0 while consumer 1 subscribed to partition 1.

I have used async-retry inside eachBatch to process and retry messages. When the number of retries are exceeded, I stop the consumer using consumer.stop().

To restart the stopped consumer, I have an event listener set up for Instrumentation event STOP, where a setTimeout function i defined as below

consumer.on(STOP, () => {
  setTimeout(async () => {
    await connectToKafka(); // consumer.connect
    await subscribeToTopic(); //consumer.subscribe
    await runConsumer(); //consumer.run
  }, consumerRestartInterval);
});

When both consumers are processing messages and one of them stops and then restarts, both consumers connect to same partition, either 0 or 1. Rebalancing does not happen during this restart. 

Only when the consumer resolves the particular message  instead of retrying, then the rebalancing happens.

Basically what i want to do is, process a message for 'n' number of times using async-retry inside eachBatch, and if for whatever reason the message process fails, i want to stop the consumer and restart it after a given interval.

The problem is when the consumer restarts, two consumers connect to same partition. This causes both consumers to read the same messages.
Not sure if the logic i am using is wrong? can someone help me in this?

Additional question is when exactly does rebalancing happen?
